### PR TITLE
Feat/coupled serializers

### PR DESCRIPTION
**Pull request by to2x**
Basically abstracted message handling logic and put it into AbstractMessage factory.
CBOR serializer now just converts its data to json and then reuses same handling logic as JSON serializer.

Same can be applied to msgpack should we want to do so.

Another handy commit is coupling serializer / serializerType into transport factory so the client doesn't need to know about the context providing coupled parameters of certain type with correct import which was confusing.
Now we can create transport like `WebSocketTransport.withCborSerializer(url)`

### CODE REVIEW

File: lib/src/message/abstract_message.dart [hunk 1]
```diff
@@ -1,5 +1,208 @@
...
+  static AbstractMessage? fromJson(
+    Object? message,
+    AbstractMessageWithPayload Function(
+      AbstractMessageWithPayload message,
+      List<dynamic> messageData,
+      int argumentsOffset,
+    )
+        payloadInjector,
+  ) {
+    if (message is! List) {
```
Comment (mjohnsullivan) [inline lib/src/message/abstract_message.dart:32]: Is this an error state? Should we be throwing exceptions rather than returning null?

**Review by mjohnsullivan — APPROVED** [review]
Nice; are we worrying about the failed tests here, or fixed in later PR?

File: lib/src/message/abstract_message.dart (hunk 1 above)
Comment (to2x) [reply lib/src/message/abstract_message.dart:32]: The serializers returned null in original implementation so i didn't want to break it (or introduce a need to refactor all dependand components).

**Comment by to2x** [comment]
The tests were breaking before these changes as well.
This PR was one i landed before i started working on main piece of refactoring

**Pull request merged**